middleware/src/system/system.go

```diff
@@ -0,0 +1,53 @@
+//package systemenv provides functionality to get data such as open ports and services running from the system the middleware is deployed on
+package environment
+
+// SystemEnv provides some information on the system we are running on.
+type SystemEnv struct {
```

Comment:
This package seems to just hold a simple struct, with no real logic. Could it just be part of the `middleware` package?

Reply:
I agree that in the current state it does not hold much logic, but this will change a lot. This package is supposed to hold and update system information as needed. This functionality does not quite fit the description of package middleware, whose job is to route data from and to services. The changes in this pull request are not meant to represent what package system will do in the end, but rather split functionality off again. I'll add  a TODO where appropriate to indicate this.

Reply:
Okay, that makes sense.

Then, if we keep this as a separate package (adding a `TODO` sounds good), maybe `package system` and `type Environment struct {...}` would be more natural, and avoid the "stuttering" (i.e names like `foo.Foo`, in Go parlance) by leading to calling code specifying a `system.Environment{}`.

The package-level comment could probably also be made more specific / hint towards the future job responsibility of the package, specifically what logic it's expected to hold.

Reply:
I think together with the TODO further down, the description is fine for now. It can grown once it contains more logic.